crates/oxc_semantic/src/builder.rs

```diff
@@ -239,6 +239,8 @@ impl<'a> SemanticBuilder<'a> {
         excludes: SymbolFlags,
     ) -> SymbolId {
         if let Some(symbol_id) = self.check_redeclaration(scope_id, span, name, excludes, true) {
+            // Push in the includes files
```

Comment:
Need to explain why this is needed, not what this is doing.

Reply:
Is this overriding the declared symbol? We should check on how typescript does this in binder.ts.

Reply:
This is a weak copy of typescript symbol declaration merging - I will comment this better thanks!

Reply:
Its not overriding the symbol its just adding flags in TS if you write code
```ts
const foo = 3
type foo = number
```
there exists only 1 symbol - that has flags specifying both type and value

Reply:
Here's a little explorer program - that can be modified that might be helpful just in general as a playground?
```ts
import * as ts from "typescript"

function createProgramFromSource(s: string) {
    const name = "dummy.ts"
    const source = ts.createSourceFile(name, s, ts.ScriptTarget.ES2015, true, ts.ScriptKind.TS)
    const host = ts.createCompilerHost({}, true)
    const old = host.getSourceFile
    host.getSourceFile = (fileName, ...rest) => {
        if (fileName === name) {
            return source}
        return old(fileName, ...rest)
    }
    const program = ts.createProgram({
        host: host,
        rootNames: [name],
        options: {}
    })
    return [program, program.getTypeChecker(), source] as const
}

const [program, checker, source] = createProgramFromSource(`
type foo = 3
const foo = 3
`)
function linter(source: ts.SourceFile) {
    const seenSymbols = new Set()
    helper(source)

    console.log(`Symbol count: ${seenSymbols.size}`)
    function helper(node: ts.Node) {

        const symbol = checker.getSymbolAtLocation(node)
        if (symbol) {
            console.log(`${node.getText()}: ${symbol.escapedName}, ${symbol.flags}`)
            seenSymbols.add(symbol)
        }
        ts.forEachChild(node, helper)
    }
}
linter(source)

```

Reply:
I see you are a TypeScript expert. So the rule of thumb is to add tons of reference, documentation and intention so the next person can understand what's going on. 

So complicated code like this, we'll need some unit tests as well to pass the knowledge forward.

Reply:
I am so far from a typescript expert but I really appreciate the kind words - especially from the author of oxc! I will pick this back up tomorrow morning with better documentation and implementation scope.